### PR TITLE
Add Annotation and Link as classes in the trace API.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Change Span to accept Links and Annotations.

This change allows to expose later an interface to export SpanData without depending on protos which is mandatory when using Stackdriver tracer.

### CODE REVIEW

File: core/src/main/java/com/google/instrumentation/trace/Attributes.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
  */
 @Immutable
 public final class Attributes {
```
Comment (adriancole) [inline core/src/main/java/com/google/instrumentation/trace/Attributes.java]: no auto-value?

File: core/src/main/java/com/google/instrumentation/trace/Span.java [hunk 2]
```diff
@@ -73,53 +73,60 @@ protected Span(SpanContext context, @Nullable EnumSet<Options> options) {
...
    */
   public abstract void addAttributes(Attributes attributes);
 
   /**
    * Adds an annotation to the {@code Span}.
    *
-   * @param description The description of the annotation time event.
+   * @param description the description of the annotation time event.
    */
   public abstract void addAnnotation(String description);
```
Comment (adriancole) [inline core/src/main/java/com/google/instrumentation/trace/Span.java]: is this the same as attribute with a description and empty attributes? If so, you could make this a concrete, but overridable method.

File: core/src/main/java/com/google/instrumentation/trace/Attributes.java (hunk 1 above)
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/trace/Attributes.java]: I think public API should not be auto-generated.

File: core/src/main/java/com/google/instrumentation/trace/Span.java (hunk 2 above)
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/trace/Span.java]: I can do that separately.

**Pull request merged**